### ISSUE TITLE
-Wshadow tripped in solvers/NonlinearProgramSnoptmex.cpp

### ISSUE THREAD

**Issue by jamiesnape** [RobotLocomotion/drake#2915]
```
Sorry, you have not run 'make' yet in the drake root,
which means you do not have the compiled MEX files needed to run this program.
Running configure and make in the drake root directory will fix this.
```
https://drake-jenkins.csail.mit.edu/view/Nightly/job/mac-clang-nightly-matlab/115/console

**Comment by david-german-tri** [comment]
Link?

**Comment by jamiesnape** [comment]
Added.

**Comment by david-german-tri** [comment]
Specifically, this looks like the first red nightly.  Odd that it includes no CMake changes.

https://drake-jenkins.csail.mit.edu/view/Nightly/job/mac-clang-nightly-matlab/112/

The changes to drake-ci on July 22 also don't smell like culprits.  Interesting.

https://github.com/RobotLocomotion/drake-ci/commits/master

**Comment by mwoehlke-kitware** [comment]
It appears that the real issue is in [`solvers/NonlinearProgramSnoptmex.cpp`](https://github.com/RobotLocomotion/drake/commits/master/drake/solvers/NonlinearProgramSnoptmex.cpp).

This also tripped CI in #2896. Not sure what changed that this is suddenly tripping, but it's a real build error and should probably be fixed by someone more familiar with the code in question than myself...

**Comment by david-german-tri** [comment]
@rpoyner-tri FYI

**Comment by mwoehlke-kitware** [comment]
Probably broke here: f2c4d53d0ed636cf482fe23e4ca0775e1a7e2333.

**Comment by david-german-tri** [comment]
Good catch, @mwoehlke-kitware.  I guess it's a clang-specific error, since it didn't show up in Linux+gcc+MATLAB.

**Comment by mwoehlke-kitware** [comment]
I believe `-Wshadow` is not enabled under GCC; see [`drake/CMakeLists.txt:83`](https://github.com/robotlocomotion/drake/blob/830d4766359c557e24cb5519e82ba8e39af353a3/drake/CMakeLists.txt#L83)

**Comment by david-german-tri** [comment]
@rpoyner-tri, over to you.  Looks like Mac MATLAB tests should be requested from Jenkins for `-Wshadow` cleanup.

**Comment by rpoyner-tri** [comment]
@jamiesnape: help me out here. I've submitted #2920 and requested a mac clang matlab build, but I see Jenkins telling me that all 'mac' nodes are offline. Is that something that should worry me?

**Comment by jamiesnape** [comment]
That is normal. They need time to boot and establish a connection to Jenkins master (2-3 minutes).

**Comment by rpoyner-tri** [comment]
closed via #2920.
